### PR TITLE
Refactor operation calls: Replace operation::run() with operation::launch_op()

### PR DESCRIPTION
**Pull request by namhyung-kim-moreh**
use ```operation::launch_op``` instead of ```oeration::run``` to support async execution in ttnn runtime.

### CODE REVIEW

File: tt_eager/tt_dnn/op_library/moreh_groupnorm_backward/moreh_groupnorm_backward_op.cpp [hunk 1]
```diff
@@ -123,13 +123,35 @@ Tensor moreh_groupnorm_backward_input_grad(
...
+
+    return output_tensors.at(0);
+
+    // return operation::run(
+    //            MorehGroupNormBackwardInputGrad{
+    //                .num_groups = num_groups, .input_grad_mem_config = std::move(input_grad_mem_config)},
+    //            {output_grad, input, mean, rstd},
+    //            {gamma},
+    //            {input_grad})
+    //     .at(0);
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_groupnorm_backward/moreh_groupnorm_backward_op.cpp:154]: Please remove it.

File: tt_eager/tt_dnn/op_library/moreh_layernorm/moreh_layernorm_op.cpp [hunk 2]
```diff
@@ -497,12 +502,34 @@ Tensor moreh_layernorm(
...
+
+    return output_tensors.at(0);
+
+    // return operation::run(
+    //            MorehLayerNorm{
+    //                .normalized_dims = normalized_dims, .eps = eps, .output_mem_config =
+    //                std::move(output_mem_config)},
+    //            {input},
+    //            {gamma, beta, mean, rstd})
+    //     .at(0);
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_layernorm/moreh_layernorm_op.cpp:532]: Please remove it.

File: tt_eager/tt_dnn/op_library/moreh_layernorm/moreh_layernorm_op.cpp [hunk 3]
```diff
@@ -520,12 +547,33 @@ Tensor moreh_layernorm(
...
+        {gamma, beta, mean, rstd});
+
+    return output_tensors.at(0);
+
+    // return operation::run_with_autoformat(
+    //            operations::primary::MorehLayerNorm{
+    //                .normalized_dims = normalized_dims, .eps = eps, .output_mem_config = std::move(output_mem_config)},
+    //            {input},
+    //            {gamma, beta, mean, rstd})
+    //     .at(0);
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_layernorm/moreh_layernorm_op.cpp:576]: Please remove it.

File: tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp [hunk 4]
```diff
@@ -96,20 +101,43 @@ operation::ProgramWithCallbacks MorehAdam::create_program(
...
+                    .output_mem_config = mem_config},
+                input_tensors,
+                optional_input_tensors,
+                optional_output_tensors);
+        },
         {param, grad, exp_avg, exp_avg_sq},
+        dummy_output_tensors,
         {max_exp_avg_sq});
 
+    std::vector<std::variant<Tensor, char*>> outputs{nullptr, nullptr, nullptr, nullptr, nullptr};
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_adam/moreh_adam_op.cpp:140]: Could you change return type of ```moreh_adam```?
```[[maybe_unused]] std::vector<std::variant<Tensor, char*>>``` -> ```std::vector<std::optional<Tensor>>```

like following link
https://github.com/tenstorrent/tt-metal/blob/5ed51dc4439e50a493a503bde5f9e59da35b55ca/tt_eager/tt_dnn/op_library/moreh_groupnorm/moreh_groupnorm_op.hpp#L71

File: tt_eager/tt_dnn/op_library/moreh_layernorm_backward/moreh_layernorm_backward_op.cpp [hunk 5]
```diff
@@ -194,8 +228,23 @@ operation::ProgramWithCallbacks MorehLayerNormBackwardGammaBetaGrad::create_prog
...
+    //     {gamma_grad, beta_grad});
+
+    // if (gamma_grad.has_value()) {
+    //     outputs[0] = gamma_grad.value();
+    // }
+    // if (beta_grad.has_value()) {
+    //     outputs[1] = beta_grad.value();
+    // }
+
+    // return outputs;
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_layernorm_backward/moreh_layernorm_backward_op.cpp:247]: Please remove it.
And
Could you change return type of ```moreh_layernorm_backward_gamma_beta_grad```?
```[[maybe_unused]] std::vector<std::variant<Tensor, char*>>``` -> ```std::vector<std::optional<Tensor>>```

File: tt_eager/tt_dnn/op_library/moreh_layernorm_backward/moreh_layernorm_backward_op.cpp [hunk 6]
```diff
@@ -156,14 +156,36 @@ operation::ProgramWithCallbacks MorehLayerNormBackwardGammaBetaGrad::create_prog
...
+
+    // return output_tensors.at(0);
+    // Inplace
+    // operation::run(
+    //     MorehLayerNormBackwardInputGrad{
+    //         .normalized_dims = normalized_dims, .output_mem_config = std::move(output_mem_config)},
+    //     {output_grad, input, mean, rstd, input_grad},
+    //     {gamma});
+
+    // return input_grad;
```
Comment (seunghwan100) [inline tt_eager/tt_dnn/op_library/moreh_layernorm_backward/moreh_layernorm_backward_op.cpp:188]: Please remove it.
And
```[[maybe_unused]] Tensor moreh_layernorm_backward_input_grad``` -> ```Tensor moreh_layernorm_backward_input_grad```

File: tt_eager/tt_lib/.rpath_checked__C
Comment (dongjin-na) [inline tt_eager/tt_lib/.rpath_checked__C:1]: Is it required?

Comment (namhyeong-kim) [reply tt_eager/tt_lib/.rpath_checked__C:1]: no. I will delete it

**Review by hschoi4448 — APPROVED** [review]

File: tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_norm.py [hunk 7]
```diff
@@ -139,6 +138,9 @@ def test_moreh_norm(input_shape, p, dim_rtol_atol, device):
 
     dim, rtol, atol = dim_rtol_atol
 
+    if dim in (None, [], [0, 1, 2, 3]) and p == 2.5:
+        pytest.skip("TODO: Check why comp_allclose result is poor on WH_B0.")
+
```
Comment (seunghwan100) [inline tests/tt_eager/python_api_testing/unit_testing/misc/test_moreh_norm.py:143]:
```suggestion
    from models.utility_functions import is_wormhole_b0
    if dim in (None, [], [0, 1, 2, 3]) and p == 2.5 and is_wormhole_b0():
        pytest.skip("TODO: Check why comp_allclose result is poor on WH_B0.")

```

**Review by seunghwan100 — APPROVED** [review]

**Pull request merged**